test/README.md

```diff
@@ -1,22 +1,119 @@
-# Testing CNI plugin with Istio
+# Istio CNI Plugin Testing
 
-This page describes the testing strategy for the CNI plugin and how to run the tests.
+This page describes the testing strategy for the Istio CNI plugin and how to run the tests.
 
 ## Types of tests
 
-The CNI will have the following tests available.
+The CNI has the following tests available.
 
-1. Unit tests
+1. Unit, Functional, and lint tests
 
-2. Manually triggered functional tests (without any Istio components)
+1. Istio e2e tests triggered on a istio/cni change
 
-3. Istio e2e tests triggered on a istio/cni change
+1. Istio e2e tests with the CNI enabled triggered on a istio/istio change
 
-4. Istio e2e tests with the CNI enabled triggered on a istio/istio change
+1. Manually executed e2e tests
 
-5. Manually executed e2e tests
+## Test Environments
 
-## Manually executed e2e tests
+The primary test environments for testing the Istio CNI plugin are:
+
+1. CircleCI
+1. Istio Prow (GKE)
+1. Developer private test environments
+
+### CircleCI Testing
+
+The "istio/cni" CircleCI test environment is under the CircleCI Istio organization.  It
+consists of 2 workflows:
+
+1. all:  PR events on the "istio/cni" github repository
+1. nightly:  triggers every night at 09:45:00 UTC
+
+All CI tests are run by both workflows.  The tests include the unit, functional, and
+lint tests as well as an e2e test.
+
+#### CircleCI e2e Tests
+
+The Istio CNI e2e_dind CircleCI job is patterned on the KDC CircleCI jobs.  The
+e2e_dind job creates a 3-node Kubernetes cluster via docker-in-docker (DinD)
+using the [kubeadm-dind-cluster (KDC)](https://github.com/kubernetes-sigs/kubeadm-dind-cluster)
+project.  Each node in the Kubernetes cluster is actually a docker container
+hosted on the CircleCI job's "remote" docker instance.  In order to run the
+Istio e2e_simple tests against that environment, a e2e_runner container is
+run within the same network namespace as the kubernetes master node
```

Comment:
kubernetes should be capitalized this is in multiple places and you should probably search for an istio vs. Istio